drivers/net/ethernet/renesas/rswitch_tc_flower.c

```diff
@@ -306,6 +294,73 @@ static int rswitch_tc_flower_replace(struct net_device *dev,
 		}
 	}
 
+	if (addr_type == FLOW_DISSECTOR_KEY_IPV6_ADDRS) {
+		struct flow_match_ipv6_addrs match;
+
+		flow_rule_match_ipv6_addrs(rule, &match);
+
+		/*
+		 * Approach is the same as for MAC addresses:
+		 * - 2 four-byte filters in expand mode for exact IPv6
+		 * - 4 four-byte filter in mask mode for masked IPv6
+		 */
+		if (!rswitch_ipv6_all_zero(&match.mask->src)) {
+			if (rswitch_ipv6_all_set(&match.mask->src)) {
+				pr_err("setting IPv6 src filter\n");
+				rc = rswitch_init_expand_pf_entry(&pf_param, PF_FOUR_BYTE,
+						be32_to_cpu(match.key->src.s6_addr32[0]),
+						be32_to_cpu(match.key->src.s6_addr32[1]),
+						RSWITCH_IPV6_SRC_OFFSET);
+				if (rc)
+					goto free;
+
+				rc = rswitch_init_expand_pf_entry(&pf_param, PF_FOUR_BYTE,
+						be32_to_cpu(match.key->src.s6_addr32[2]),
+						be32_to_cpu(match.key->src.s6_addr32[3]),
+						RSWITCH_IPV6_SRC_OFFSET + 8);
+				if (rc)
+					goto free;
+			} else {
+				for (i = 0; i < 4; i++) {
+					rc = rswitch_init_mask_pf_entry(&pf_param, PF_FOUR_BYTE,
```

Comment:
For example, we can add check: if mask contains all '1', we can omit pf_setup

Reply:
Yeah, right. But mask should be all zero to omit pf_setup. Done, thanks for cool suggestion.